lib/Doctrine/Common/Proxy/AbstractProxyFactory.php

```diff
@@ -216,6 +226,13 @@ private function getProxyDefinition($className)
                     require $fileName;
                     break;
 
+                case self::AUTOGENERATE_FILE_OUTDATED:
+                    if ( ! file_exists($fileName) || filemtime($classMetadata->getReflectionClass()->getFileName()) > filemtime($fileName)) {
```

Comment:
It should probably re-generate it
On Dec 12, 2015 15:54, "Konstantin Myakshin" <notifications@github.com>
wrote:

> In lib/Doctrine/Common/Proxy/AbstractProxyFactory.php
> <https://github.com/doctrine/common/pull/702#discussion_r47432449>:
>
> > @@ -216,6 +226,13 @@ private function getProxyDefinition($className)
> >                      require $fileName;
> >                      break;
> >
> > +                case self::AUTOGENERATE_FILE_OUTDATED:
> > +                    if ( ! file_exists($fileName) || filemtime($classMetadata->getReflectionClass()->getFileName()) > filemtime($fileName)) {
>
> I haven't know that proxy generation can be used for internal classes. In
> this case we shouldn't generate new proxy class. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/doctrine/common/pull/702/files#r47432449>.
>